src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -693,6 +700,85 @@ protected override void SetItem(int index, string item)
             {
                 base.SetItem(index, string.IsNullOrEmpty(item) || item == "*.*" ? "*" : item);
             }
+
+            internal string[] GetFilters() => ((ImmutableStringList)Items).Items;
+
+            /// <summary>
+            /// List that maintains its underlying data in an immutable array, such that the list
+            /// will never modify an array returned from its Items property. This is to allow
+            /// the array to be enumerated safely while another thread might be concurrently mutating
+            /// the collection.
+            /// </summary>
+            private sealed class ImmutableStringList : IList<string>
```

Comment:
Is this potentially useful in other places? Would it be useful to unnest and make internal so we can unit test now / share in the future?

Reply:
> Would it be useful to unnest and make internal so we can unit test now / share in the future?

I think that's fine if/when we need it elsewhere.  In the meantime, I've explicitly not implemented all of the functionality, only what's needed.